beer-garden/functionary
# #93 - Further refine results output formatting

Closes #93 

This PR builds on the previous work for displaying nicely formatted JSON results and table output where available.  It has two main goals:

* To have the /tasks/{id}/result/ API show the results as the correct type, rather than always being stringified JSON.  Objects now return as objects, numbers as numbers, etc.
* To no longer require explicitly declaration of the output format up front on the function definition.  It was previously the case that when viewing the results you would be shown only a the raw JSON or a table, based on what output format had been declared.  Now you can toggle between both formats when appropriate like this:

### JSON Raw Output
![image](https://user-images.githubusercontent.com/86254668/198389875-be18f13a-97f3-41d8-8e2f-2ff1f6a9987d.png)

### JSON Table Output
![image](https://user-images.githubusercontent.com/86254668/198389970-238773a4-2d27-4b1f-af85-6dc3648b4e7e.png)

## Other Notes

* I removed the `output_format` field from the Function and Task model.  It turned out we can do everything we want to without it.
* I also added a permissions check to the unicorn view, as it was possible for data to be accessed through the unicorn endpoint by any logged in user before, regardless of their actual permissions.

## Testing Instructions

* Both the python and javascript templates had to be updated to support these changes.  Be sure to run the `./build.sh -p` from the templates folder to get the new versions of the templates in your registry.
* I added a simple "demo" example package that has useful functions for testing the UI rendering of JSON and text (including CSV formatted text).  Publish that package to an environment (**after publishing the new templates**) to ease your testing.
* Task the output_json and output_text functions with data that can and cannot be rendered as a table. The descriptions of each function explain what should generate table output.
* Retrieve the task results via the API as well and verify that the date type of the `result` property is what you expect based on the return type of the function.
